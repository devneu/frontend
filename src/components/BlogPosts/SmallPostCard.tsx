import { Button, Card, Typography } from 'antd';
import { ArrowRightOutlined } from '@ant-design/icons';
import React from 'react';
import './SmallPostCard.scss';

const { Title, Paragraph } = Typography;

interface SmallPostCardProps {
  post: {
    postImg: string;
    postDate: string;
    postTitle: string;
    postText: string;
  };
}

// eslint-disable-next-line no-redeclare
const SmallPostCard = ({ post }: SmallPostCardProps) => (
  <Card className="post-card" cover={<img alt="card-img" src={post.postImg} />}>
    <Card.Meta description={post.postDate} />
    <Title level={2}>{post.postTitle}</Title>
    <Paragraph className="post-text">{post.postText}</Paragraph>
    <Button type="link" href="#" className="button-card">
      Read more
      <ArrowRightOutlined />
    </Button>
  </Card>
);

export default SmallPostCard;
