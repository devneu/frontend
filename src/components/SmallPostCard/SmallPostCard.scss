.post-card {
  border-radius: 5px;
  margin: 10px 0px;

  img {
    border-radius: 5px 5px 0px 0px;
  }

}