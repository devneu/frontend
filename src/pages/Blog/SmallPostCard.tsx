import {Button, Card, Col, Typography} from "antd";
import {ArrowRightOutlined} from "@ant-design/icons";
import React from "react";

const { Title, Text } = Typography;
const { Meta } = Card;

interface SmallPostCard {
    postImg: string,
    postTitle: string,
    postText: string,
    sizeBlock: number
}


const SmallPostCard = ({postImg, postTitle, postText, sizeBlock}: SmallPostCard) => {

    return (
        <Col span={sizeBlock}>
            <Card
                className="post-card"
                cover={
                    <img alt="card-img"
                         src={postImg}
                    />
                }>
                <Meta description="January 1, 2021"/>
                <Title level={2}>{postTitle}</Title>
                <Text className="post-text">{postText}</Text>
                <br/>
                <Button type="primary" className="button-card">Read more<ArrowRightOutlined /></Button>
            </Card>
        </Col>
    );
}

export default SmallPostCard;