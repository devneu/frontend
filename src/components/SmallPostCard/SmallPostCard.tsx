import { Button, Card, Typography } from 'antd';
import React from 'react';
import { ArrowRightOutlined } from '@ant-design/icons';
import { NavLink } from 'react-router-dom';
import { Post } from '../../types/Post.interface';
import './SmallPostCard.scss';

interface SmallPostCardProps {
    post: Post;
}

const SmallPostCard = ({ post }: SmallPostCardProps) => (
  <Card className="post-card" cover={<img alt="card-img" src={post.img} />}>
    <Card.Meta description={post.date} />
    <Typography.Title level={2}>{post.title}</Typography.Title>
    <Typography.Paragraph className="post-text">{post.text}</Typography.Paragraph>
    <NavLink to={`/blog/${post.id}`}>
      <Button type="primary">
        Read more
        <ArrowRightOutlined />
      </Button>
    </NavLink>
  </Card>
);

export default SmallPostCard;
