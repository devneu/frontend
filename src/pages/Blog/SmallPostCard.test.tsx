import renderer from 'react-test-renderer';
import SmallPostCard from './SmallPostCard';

describe('SmallPostCard', () => {
    it('Should successfully render', () => {
        const wrapper = renderer
            .create(
                <SmallPostCard postImg="post image" postTitle="post title" postText="post text" sizeBlock={12}/>
            )
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});
