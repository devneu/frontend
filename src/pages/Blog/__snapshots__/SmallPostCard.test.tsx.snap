// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallPostCard Should successfully render 1`] = `
<div
  className="ant-col ant-col-12"
  style={Object {}}
>
  <div
    className="ant-card ant-card-bordered post-card"
  >
    <div
      className="ant-card-cover"
    >
      <img
        alt="card-img"
        src="post image"
      />
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-card-meta"
      >
        <div
          className="ant-card-meta-detail"
        >
          <div
            className="ant-card-meta-description"
          >
            January 1, 2021
          </div>
        </div>
      </div>
      <h2
        className="ant-typography"
        style={
          Object {
            "WebkitLineClamp": undefined,
          }
        }
      >
        post title
      </h2>
      <span
        className="ant-typography post-text"
        style={
          Object {
            "WebkitLineClamp": undefined,
          }
        }
      >
        post text
      </span>
      <br />
      <button
        className="ant-btn ant-btn-primary button-card"
        onClick={[Function]}
        type="button"
      >
        <span>
          Read more
        </span>
        <span
          aria-label="arrow-right"
          className="anticon anticon-arrow-right"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="arrow-right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 000-48.4z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
